### PR TITLE
Add missing explicit scopes to Omniauth adapters

### PR DESCRIPTION
**Pull request by fdocr**
## What type of PR is this? (check all applicable)

- [x] Optimization

## Description

In our OAuth providers we didn't specify all of their scopes explicitly. This wasn't an issue because the underlying Omniauth gems provide a default scope, but it was brought up to our attention that it's better to explicitly specify our scopes for security reasons.

There shouldn't be any real differences in how the code will work, we're just explicitly requesting the scopes we need instead of relying in the upstream gem default scopes.

## Related Tickets & Documents

- Closes https://github.com/forem/forem-internal-eng/issues/459

## QA Instructions, Screenshots, Recordings

I ran through the updated providers and didn't find any issues

1. Enable [Google](https://developers.forem.com/backend/auth-google)/[Facebook](https://developers.forem.com/backend/auth-facebook) OAuth flows
2. Sign in/up using those providers
3. Shouldn't find any problems with step 2

https://user-images.githubusercontent.com/6045239/170286790-d1e6358c-ecc0-4e5c-9d7d-c81b1251ce49.mov

### UI accessibility concerns?

None

## Added/updated tests?

- [x] No, and this is why: We didn't test the exact scopes requested to OAuth providers before and I don't think we need to IMO

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams

## [optional] What gif best describes this PR or how it makes you feel?

man saying: put it in writing

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -49,7 +51,7 @@
   # Configure the e-mail address which will be shown in Devise::Mailer,
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = "#{ENV['COMMUNITY_NAME']} <#{ENV['DEFAULT_EMAIL']}>"
+  config.mailer_sender = "#{ENV.fetch('COMMUNITY_NAME', nil)} <#{ENV.fetch('DEFAULT_EMAIL', nil)}>"
```
Comment (fdocr) [inline config/initializers/devise.rb:54]: This appears to be a Rubocop automatic fix

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -13,25 +13,27 @@
 end
 
 GITHUB_OMNIAUTH_SETUP = lambda do |env|
+  env["omniauth.strategy"].options[:scope] = "user:email"
   env["omniauth.strategy"].options[:client_id] = Settings::Authentication.github_key
   env["omniauth.strategy"].options[:client_secret] = Settings::Authentication.github_secret
-  env["omniauth.strategy"].options[:scope] = "user:email"
 end
```
Comment (fdocr) [inline config/initializers/devise.rb:19]: This one was already here but I moved it up so all of the providers here have the scope appear first (consistency/readability)

**Review by maestromac — APPROVED** [review]
Makes sense. LGTM!

**Review by Ridhwana — APPROVED** [review]
Looks good to me!

**Pull request merged**